java-checks/src/main/java/org/sonar/java/checks/PublicStaticFieldShouldBeFinalCheck.java

```diff
@@ -66,7 +67,8 @@ public void visitClass(ClassTree tree) {
   private boolean isPublicStaticNotFinal(VariableTree tree) {
     boolean isPublic = false, isStatic = false, isFinal = false;
 
-    for (Modifier modifier : tree.modifiers().modifiers()) {
+    for (ModifierKeywordTree modifierKeywordTree : tree.modifiers().modifiers()) {
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Impossible cast from org.sonar.plugins.java.api.tree.Modifier to org.sonar.plugins.java.api.tree.ModifierKeywordTree in org.sonar.java.checks.PublicStaticFieldShouldBeFinalCheck.isPublicStaticNotFinal(VariableTree) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ABC_IMPOSSIBLE_CAST)